extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-preview.js

```diff
@@ -0,0 +1,654 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AmpStoryPlayer} from '../../../src/amp-story-player/amp-story-player-impl';
+import {CSS} from '../../../build/amp-story-dev-tools-tab-preview-0.1.css';
+import {closest} from '../../../src/dom';
+import {escapeCssSelectorIdent} from '../../../src/css';
+import {htmlFor} from '../../../src/static-template';
+import {setStyles} from '../../../src/style';
+import {toArray} from '../../../src/types';
+import {updateHash} from './utils';
+
+/**
+ * Creates a tab content, will be deleted when the tabs get implemented.
+ * @param {!Window} win
+ * @param {string} storyUrl
+ * @param {?string} devices
+ * @return {!Element} the layout
+ */
+export function createTabPreviewElement(win, storyUrl, devices) {
+  const element = win.document.createElement('amp-story-dev-tools-tab-preview');
+  element.setAttribute('story-url', storyUrl);
```

Comment:
here and below, please use data-attributes

Reply:
Tried to think of the `amp-story-dev-tools-tab-preview` as an independent component that could be instantiated by themselves. Don't we have most attributes as regular attributes (without `data-`) on other components? Or why would we store these attributes as `data-`?

Reply:
We usually use attributes without `data-` because they're invalid by default, and we can control them through validation. `data-` are valid AMP by default, but they're also valid HTML.
Because we don't really care about AMP validation here, I think we should go with valid HTML and use `data-`